GitCommands/Settings/AppSettings.cs

```diff
@@ -141,6 +141,8 @@ internal static void SetDocumentationBaseUrl(string currentGitBranch)
                 }
             }
 
+            // override for release, Git branch is still release/4.0
+            docVersion = "en/release-4.2/";
```

Comment:
I'm happy to take it as a quick fix, but this can be generalised using `AppSettings.ProductVersion`:
https://github.com/gitextensions/gitextensions/blob/0898ee687cf676be0d3d4bbc190447b036267043/GitCommands/Settings/AppSettings.cs#L20

Reply:
Since testcases for `SetDocumentationBaseUrl_should_currectly_append_verison` and `GetUrl` are broken, we should switch to `ProductVersion` now.
You or me, @gerhardol?